README.md

```diff
@@ -223,6 +237,22 @@ Example of customizing semantic colors in settings.json:
 
     -   Convert import format according to `python.analysis.importFormat`.
 
+# Auto/Add import and indexing
+
+Pylance has a feature called ```auto import``` which shows symbols from installed packages and user files during completion. When committed, pylance will automatically generate an ```import``` statement. It also has something called ```add import``` which shows symbols from installed packages and user files for an unknown identifier. When selected, this will also insert an ```import``` statement.
+
+Since these features can be costly, we have enabled the minimum set and provide multiple settings to control the behavior.
+
+When ```python.analysis.indexing``` is off, we will only show symbols already used by opened files transitively. If ```indexing``` is on, we will spend some time indexing installed packages and user files when pylance is started, but after that, we will provide symbols that you haven't used yet in completion and code action.
+
+But since indexing is an expensive operation, we do not index every symbol from all installed packages but only top-level symbols. In another words, symbols in ```__all__``` in ```package/__init__```. If users want more than that, they need to use ```python.analysis.packageIndexDepths``` to tell us which packages they want us to spend more time on. Symbols indexed will show up both in completion and code actions.
+
+For user files, we will index up to 2000 files. One difference between user files and installed packages is that symbols from user files won't appear in completion unless they are already used. Unlike installed packages, we, by default, index all symbols from user files, which can cause thousands of user symbols to show up in completion. 
```

Comment:
It's not clear that the last sentence is explaining what would happen if we didn't limit user symbols to those already in use. Maybe something like this?
```suggestion
We will index up to 2000 user files, and unlike installed packages we index all symbols in user files. Symbols from user files won't appear in completions unless they are already used, otherwise the large number of symbols would make the completion list unmanageable. 
```